ethereum/contracts/IncentivizedOutboundChannel.sol

```diff
@@ -19,6 +31,28 @@ contract IncentivizedOutboundChannel is OutboundChannel {
         override
     {
         nonce = nonce + 1;
-        emit Message(msg.sender, nonce, payload);
+
+
+        emit Message(msg.sender, nonce, payload, gasFee);
+    }
+
+    modifier onlyFeeController {
+        require(msg.sender == feeController, "Caller is not a fee controller");
+        _;
+    }
+
+    /**
+    * @dev Sets gasFee. Only feeController is allowed to set gasFee
+    */
+    function setGasFee(uint256 _fee) public onlyFeeController {
+        require(_fee > 0, "fee must be positive");
+        gasFee = _fee;
+    }
+
+    /**
+    * @dev Change feeController address
+    */
+    function setFeeController(address _feeControllerAddress) public onlyFeeController {
```

Comment:
I think this contract should be `Ownable` and the modifier should be set to `onlyOwner`.

Reply:
You mean to setFeeController?

Reply:
No I mean that the contract should be `Ownable` and that the modifier should be changed from `onlyFeeController` to `onlyOwner`. Not sure what the implementation details are here, but I think that makes sense.

Reply:
feeController is different from the owner of the contract. Only feeController is allowed to set gasFee and change itself.

Reply:
I agree with @blablacio here. The `owner` is just an address that has elevated permissions. In this case, the `owner` would have permission to update fees. It's better to reuse existing secure contracts (such as OpenZeppelin's `Ownable.sol`) when possible to reduce potential attack vectors on new code.

Reply:
disagree - I don't think the terminology owner is good or reflects the meaning of this role. using existing libs for roles/authorization though, yeh that's good.

Reply:
Ah okay, that perspective makes sense. Worth looking into [AccessControl.sol](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/access/AccessControl.sol) here then.

Reply:
[ AccessControl.sol ](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/access/AccessControl.sol) also has `DEFAULT_ADMIN_ROLE` which has all the permissions (like a super user/role). It by default set to `0x00`, can this be a problem? Should we set owner of the contract as DEFAULT_ADMIN?

Reply:
That sounds fine. Either way, it may not matter much soon. One of my upcoming tasks is to introduce collective governance mechanisms. This work will likely end up refactoring any code related to on-chain configuration and permissions. so this `setFeeController` may not even be necessary right now.